SafeguardSessions/modules/schema/SchemaUtils.pqm

```diff
@@ -118,30 +117,23 @@ let
             Table.Buffer(flattened),
     SchemaUtils.Flatten = (listOfRecords as list, optional columnsToRemove as list) as table =>
         let
+            asTable = PaddedTable.FromRecords(listOfRecords),
+            reducedTable = Table.RemoveColumns(
+                asTable, if columnsToRemove <> null then columnsToRemove else {}, MissingField.Ignore
+            ),
             flattenedSubTable =
-                if SchemaUtils.IsFlat(listOfRecords) then
-                    let
-                        asTable = PaddedTable.FromRecords(listOfRecords),
-                        reducedTable = Table.RemoveColumns(
-                            asTable, if columnsToRemove <> null then columnsToRemove else {}, MissingField.Ignore
-                        )
-                    in
-                        Table.Buffer(reducedTable)
+                if SchemaUtils.IsFlat(reducedTable) then
+                    Table.Buffer(reducedTable)
                 else
                     let
-                        // Remove columns
-                        asTable = Table.Buffer(PaddedTable.FromRecords(listOfRecords)),
-                        reducedTable = Table.RemoveColumns(
-                            asTable, if columnsToRemove <> null then columnsToRemove else {}, MissingField.Ignore
-                        ),
                         // Flatten lists
-                        listFieldNames = SchemaUtils.FieldNamesOfType(listOfRecords, List.Type),
+                        listFieldNames = SchemaUtils.FieldNamesOfType(reducedTable, List.Type),
                         withFlattenedLists = SchemaUtils.FlattenLists(reducedTable, listFieldNames),
                         // Flatten records
-                        recordFieldNames = SchemaUtils.FieldNamesOfType(listOfRecords, Record.Type),
+                        recordFieldNames = SchemaUtils.FieldNamesOfType(reducedTable, Record.Type),
                         withFlattenedRecords = SchemaUtils.FlattenRecords(withFlattenedLists, recordFieldNames)
                     in
-                        @SchemaUtils.Flatten(Table.ToRecords(withFlattenedRecords))
+                        @SchemaUtils.Flatten(Table.ToRecords(withFlattenedRecords), columnsToRemove)

```

Comment:
Nice modification!

There is only a small thing that we need to take into account during testing/usage: on lower levels the names are detemined by https://github.com/OneIdentity/SafeguardPowerBI/blob/main/SafeguardSessions/modules/schema/SchemaUtils.pqm#LL108C21-L108C29